shell/platform/windows/win32_dpi_helper.cc

```diff
@@ -13,60 +13,56 @@ bool AssignProcAddress(HMODULE comBaseModule, const char* name, T*& outProc) {
 }  // namespace
 
 Win32DpiHelper::Win32DpiHelper() {
-  // TODO ensure that this helper works correctly on downlevel builds.
   user32_module_ = LoadLibraryA("User32.dll");
   if (user32_module_ == nullptr) {
     return;
   }
-
-  if (!AssignProcAddress(user32_module_, "EnableNonClientDpiScaling",
-                         enable_non_client_dpi_scaling_)) {
+  shlib_module_ = LoadLibraryA("Shcore.dll");
+  if (shlib_module_ == nullptr) {
     return;
   }
 
-  if (!AssignProcAddress(user32_module_, "GetDpiForWindow",
-                         get_dpi_for_window_)) {
-    return;
+  if (AssignProcAddress(user32_module_, "GetDpiForWindow",
+                        get_dpi_for_window_)) {
+    dpi_for_window_supported_ = true;
```

Comment:
We can early return (with a comment) here, since we don't need the later functions if we have the per-window API, right?

If so, we should also move the Shcore.dll load down to after the early return.